### PR TITLE
[API Server] Add Ray Job output - start/end time and ray cluster name

### PR DESCRIPTION
**Pull request by han-steve**
## Why are these changes needed?

Adding the starttime, endtime, and the rayclustername fields to the api server output to match the status fields on the rayjob CRD. 

These fields are useful for displaying the job start/end time information and easily pulling the cluster information. For us, they are also useful for generating the grafana links to show the Prometheus metrics collected during the job's run. 

## Checks

- [x] I've made sure the tests are passing.
- Testing Strategy
   - [x] Unit tests
   - [x] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: proto/go_client/config.pb.gw.go [hunk 1]
```diff
@@ -789,7 +789,7 @@ func RegisterImageTemplateServiceHandlerServer(ctx context.Context, mux *runtime
 // RegisterComputeTemplateServiceHandlerFromEndpoint is same as RegisterComputeTemplateServiceHandler but
 // automatically dials to "endpoint" and closes the connection when "ctx" gets done.
 func RegisterComputeTemplateServiceHandlerFromEndpoint(ctx context.Context, mux *runtime.ServeMux, endpoint string, opts []grpc.DialOption) (err error) {
-	conn, err := grpc.NewClient(endpoint, opts...)
+	conn, err := grpc.Dial(endpoint, opts...)
```
Comment (YQ-Wang) [inline proto/go_client/config.pb.gw.go:792]: I noticed this change last time as well. I manually put it back to the original code.

**Comment by YQ-Wang** [comment]
LGTM

**Comment by YQ-Wang** [comment]
cc @kevin85421 @andrewsykim

File: proto/go_client/config.pb.gw.go (hunk 1 above)
Comment (han-steve) [reply proto/go_client/config.pb.gw.go:792]: Oh ok... Wonder why the generated pb is inconsistent.

**Comment by kevin85421** [comment]
We are currently in KubeCon. I have already approved the CI. I will merge it if CI passes because @YQ-Wang has already approved this API server PR.

**Comment by kevin85421** [comment]
@han-steve would you mind taking a look at the lint error? Thanks!

**Comment by han-steve** [comment]
Yep!

**Comment by kevin85421** [comment]
@han-steve would you mind fixing the conflict? Thanks!

**Comment by han-steve** [comment]
Done :)

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**